include/alp/amf-based/matrix.hpp

```diff
@@ -699,14 +699,14 @@ namespace alp {
 	>
 	typename internal::new_container_type_from<
 		typename SourceMatrix::template view_type< view::gather >::type
-	>::template change_structure< structures::Diagonal >::type
```

Comment:
Is this reverting to Square? I thought the change to Diagonal was appropriate (I think we briefly touched on this during a discussion with @vdimiczrc as well?)

Reply:
I have a rectangular-diagonal structure now.